flink-clients/src/test/java/org/apache/flink/client/program/rest/RestClusterClientTest.java

```diff
@@ -909,6 +910,17 @@ void testSubmitJobAndWaitForExecutionResult() throws Exception {
         final AtomicBoolean firstSubmitRequestFailed = new AtomicBoolean(false);
         failHttpRequest =
                 (messageHeaders, messageParameters, requestBody) -> {
+                    if (messageHeaders instanceof CustomHeadersDecorator) {
```

Comment:
This is a bit hard to follow. Shouldn't we expect to find `CustomHeadersDecorator` and throw an error otherwise? If not then the bug you are trying to fix could be re-introduced w/o the test breaking?

Reply:
We are interested in `messageHeaders` itself, and it is double wrapped in `CustomHeadersDecorator(new UrlPrefixDecorator(messageHeaders))`. The test was checking for instance type of `messageHeaders`, which this code is trying to do.
`RestClusterClient.sendRequest()` function internally calls `RestClient.sendRequest()` function which accesses messageHeaders and customHeaders to construct httpRequest, and it is not accessed anywhere else. That part in RestClient is already updated to work with decorated messageHeaders:
https://github.com/apache/flink/pull/22816/files#diff-d6cdab55fefade687bd62fb086b6fa85aa4f8c91a745e8ec406fb8ac8982b9e1R440 

https://github.com/apache/flink/pull/22556/files#diff-d6cdab55fefade687bd62fb086b6fa85aa4f8c91a745e8ec406fb8ac8982b9e1R358

Reply:
My question is why the unwrapping is conditional? Isn't it an error if `messageHeaders` is not double wrapped? If so, then the test should check that.

Reply:
I see your point. I removed all checks, as now we will always wrap messageHeaders into `CustomHeadersDecorator(new UrlPrefixDecorator(messageHeaders))` before sending the request, thus we do not need those condition checks anymore.